pallets/stableswap/src/trade_execution.rs

```diff
@@ -197,4 +199,79 @@ impl<T: Config> TradeExecution<T::RuntimeOrigin, T::AccountId, T::AssetId, Balan
 			_ => Err(ExecutorError::NotSupported),
 		}
 	}
+
+	fn calculate_spot_price_with_fee(
+		pool_type: PoolType<T::AssetId>,
+		asset_a: T::AssetId,
+		asset_b: T::AssetId,
+	) -> Result<FixedU128, ExecutorError<Self::Error>> {
+		match pool_type {
```

Comment:
would be ideal to have this whole method covered by single math function so it doesn't have to be reimplemented in SDK as well